hubspot3/tickets.py

```diff
@@ -78,3 +90,181 @@ def get_all(self, limit: int = -1, **options) -> list:
             offset = batch["offset"]
 
         return output if not limited else output[:limit]
+
+    def _get_batch(self, ids: List[int], extra_properties: Union[List[str], str] = None,
+                   with_history: bool = False) -> Dict[str, dict]:
+        """given a batch of vids, get more of their info"""
+        # default properties to fetch
+        properties = set(self.__default_batch_properties)
+
+        # append extras if they exist
+        if extra_properties:
+            if isinstance(extra_properties, list):
+                properties.update(extra_properties)
+            if isinstance(extra_properties, str):
+                properties.add(extra_properties)
+        if with_history:
+            property_name = "propertiesWithHistory"
+        else:
+            property_name = "properties"
+        params = {}
+        params[property_name] = list(properties)
+        params['includeDeletes'] = True
+        # run the ids as a list of 100
+        batch = {}
+        while len(ids) > 0:
+            partial_ids = ids[:100]
+            ids = ids[100:]
+            partial_batch = self._call(
+                "objects/tickets/batch-read",
+                method="POST",
+                doseq=True,
+                params=params,
+                data={"ids": partial_ids}
+            )
+            batch.update(partial_batch)
+        return batch
+
+    def get_batch(self, ids: List[int],
+                  extra_properties: Union[List[str], str] = None) -> List[dict]:
+        """given a batch of ticket, get more of their info"""
+        batch = self._get_batch(ids, extra_properties=extra_properties, with_history=False)
+        return [prettify(batch[ticket_id], id_key="objectId") for ticket_id in batch]
+
+    def get_batch_with_history(self, ids: List[int],
+                               extra_properties: Union[List[str], str] = None) -> Dict[str, dict]:
+        """given a batch of ticket, get more of their info with history"""
+        batch = self._get_batch(ids, extra_properties=extra_properties, with_history=True)
+        return batch
+
+    def _get_recent(
+        self,
+        recency_type: str,
+        limit: int = -1,
+        vid_offset: int = 0,
+        time_offset: int = 0,
+        max_time_diff_ms: int = 60000,
+        **options
+    ) -> List:
+
+        limited = limit > 0
+        total_tickets = 0
+        finished = False
+        while not finished:
+            params = {}
+            if vid_offset:
+                params["objectId"] = vid_offset
+            if time_offset:
+                params["timestamp"] = int(time_offset)
```

Comment:
Is a timestamp an int or a datetime? Python does have a built-in datetime module. If the time can be reliably parsed here, I think taking the burden off your API's client is a bonus.

Reply:
timestamp is a int in ms